sdk/keyvault/keyvault-admin/review/keyvault-admin.api.md

```diff
@@ -0,0 +1,102 @@
+## API Report File for "@azure/keyvault-admin"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import * as coreHttp from '@azure/core-http';
+import { PagedAsyncIterableIterator } from '@azure/core-paging';
+import { TokenCredential } from '@azure/core-http';
+
+// @public (undocumented)
+export class AccessControlClient {
+    constructor(vaultUrl: string, credential: TokenCredential, pipelineOptions?: AccessControlClientOptions);
+    createRoleAssignment(scope: RoleAssignmentScope, name: string, options?: CreateRoleAssignmentOptions): Promise<KeyVaultRoleAssignment>;
+    deleteRoleAssignment(scope: RoleAssignmentScope, name: string, options?: DeleteRoleAssignmentOptions): Promise<KeyVaultRoleAssignment>;
+    getRoleAssignment(scope: RoleAssignmentScope, name: string, options?: GetRoleAssignmentOptions): Promise<KeyVaultRoleAssignment>;
+    listRoleAssignments(scope: RoleAssignmentScope, options?: ListRoleAssignmentsOptions): PagedAsyncIterableIterator<KeyVaultRoleAssignment>;
+    listRoleDefinitions(scope: RoleAssignmentScope, options?: ListRoleDefinitionsOptions): PagedAsyncIterableIterator<KeyVaultRoleDefinition>;
+    readonly vaultUrl: string;
+}
+
+// @public
+export interface AccessControlClientOptions extends coreHttp.PipelineOptions {
+    serviceVersion?: SUPPORTED_API_VERSIONS;
+}
+
+// @public
+export interface CreateRoleAssignmentOptions extends coreHttp.OperationOptions {
+    properties?: RoleAssignmentProperties;
+}
+
+// @public
+export interface DeleteRoleAssignmentOptions extends coreHttp.OperationOptions {
+}
+
+// @public
+export interface GetRoleAssignmentOptions extends coreHttp.OperationOptions {
+}
+
+// @public
+export interface KeyVaultRoleAssignment {
+    readonly id?: string;
+    readonly name?: string;
+    properties?: RoleAssignmentPropertiesWithScope;
+    readonly type?: string;
+}
+
+// @public
+export interface KeyVaultRoleDefinition {
+    assignableScopes?: string[];
+    description?: string;
+    readonly id?: string;
+    readonly name?: string;
+    permissions?: RoleDefinitionPermission[];
```

Comment:
`KeyVaultPermission` is what we have in .NET, but @christothes I'm seeing a lot of inconsistency in naming. We could probably ignore this for now but will have to figure it all out later. Maybe you and @sadasant can work it out now. The main thing is that models and clients should at least be non-generic terms (like "Definition"), and we probably want to consider we may have generic RBAC client in the future, so the "KeyVault" prefix might be a good idea universally. Curious what @KrzysztofCwalina thinks.

Reply:
Sounds good to me! I'll do the changes you suggest for now to align to .Net and then we'll follow up.

Reply:
Should the KeyVault prefix exist on the `RoleDefinition` type, for example? It seems generic, and since we're going to have other admin clients, it seemed appropriate.

Reply:
Same goes for RoleAssignment, should it be prefixed too?

Reply:
Addressed the KeyVaultPermission part of this conversation here: https://github.com/Azure/azure-sdk-for-js/pull/10815/commits/26e95837b87e1b804e309fe45421ee01ead5d165

Reply:
There's a similar discussion regarding the Key Vault identifier parser here: https://github.com/Azure/azure-sdk-for-js/pull/10736#discussion_r474359128

Perhaps we all should have a meeting? Or toss a coin? I don't want to be the one pushing to one direction or another.

Reply:
As @heaths points out, we are not quite consistent in .NET currently, but I think we should address that. What I propose is the following model names:

`KeyVaultPermission`
`KeyVaultRoleAssignment`
`KeyVaultRoleAssignmentProperties`
`KeyVaultRoleAssignmentPropertiesWithScope`
`KeyVaultRoleDefinition`

Reply:
If @heaths agrees to that, I would also agree.

Reply:
cc: @ramya-rao-a ^

Reply:
I agree with the names above. A note to @ramya-rao-a: we may have more generic names in the future. I proposed a generic API to Krzysztof a while back since we're likely to start seeing more data-plane RBAC, but - at the time, at least (last March) - we were not ready to start discussing it for Key Vault. I don't think we still are, but it's a possibility at some point, so we don't want to have name collisions. The names may be long in some cases, but the users shouldn't rarely have to type most of them (typically output models).

Reply:
The proposal by @christothes above looks good to me. It is in alignment with the guideline that a service prefix should be used if the absence of it results in a generic term with no hints as to which service it is related to.

Reply:
I'm happy to read both responses! I'll do the update.

Reply:
Note:

If `KeyVaultRoleAssignmentProperties` and `KeyVaultRoleAssignmentPropertiesWithScope` satisfies the guideline that a service prefix should be used if the absence of it results in a generic term with no hints as to which service it is related to, then probably `KeyVaultCertificateIdentifier` does too? from [this other thread](https://github.com/Azure/azure-sdk-for-js/pull/10736#discussion_r477627549).

Reply:
I've [updated this PR](https://github.com/Azure/azure-sdk-for-js/pull/10815/commits/1b0f6420a89765236d42abd8537ee777609383ac) to comply with:

```
KeyVaultPermission
KeyVaultRoleAssignment
KeyVaultRoleAssignmentProperties
KeyVaultRoleAssignmentPropertiesWithScope
KeyVaultRoleDefinition
```

Reviews appreciated ✨

Reply:
One more consistency naming issue - java is using `RoleScope` rather than `RoleAssignmentScope`. I'm thinking we should snap to that or possibly even `KeyVaultRoleScope`. Thoughts? 

//cc: @vcolin7

Reply:
I don't like the assignment part either! but otherwise I'll leave y'all to agree

Reply:
Yeah I prefer to have the `Assignment` or `Definition` removed from the scope class name, since the scope can be used with both `RoleAssignment` and `RoleDefinition`. Aside from that I don't mind using either name, although I think using a "KeyVault" prefix everywhere seems a bit overkill if the classes that are in the Key Vault package and have names that are not that widely used (i.e. `KeyVaultError` instead of `Error` looks good to me but you could argue that `KeyVaultRoleAssignment` instead of `RoleAssignment` is unnecessary).

Reply:
Reading a bit more of the conversation above I think it's a good idea to use the prefix to for the sake of consistency.

Reply:
Python here: https://github.com/Azure/azure-sdk-for-python/pull/13372 seems to be using these as well:

KeyVaultPermission
KeyVaultRoleAssignment
KeyVaultRoleAssignmentProperties
KeyVaultRoleAssignmentPropertiesWithScope
KeyVaultRoleDefinition

This should go to Archboard review eventually, but I think we could keep things consistent together first.